benchbuild/utils/run.py

```diff
@@ -5,19 +5,25 @@
 from contextlib import contextmanager
 
 import attr
+import typing_extensions as te
 from plumbum import TEE, local
 from plumbum.commands import ProcessExecutionError
+from plumbum.commands.base import BaseCommand
 
 from benchbuild import settings, signals
 
-if t.TYPE_CHECKING:
-    import mypy_extensions as mypy  # pylint: disable=unused-import
+CommandResult = t.Tuple[int, str, str]
+
+
+class WatchableCommand(te.Protocol):
+
+    def __call__(self, *args: t.Any, **kwargs: t.Any) -> CommandResult:
+        ...
```

Comment:
The ellipsis object here is a bit confusing. I had to read up if it is valid in this context, seems that it is because it does nothing, a pass would be a bit clearer :wink: 
```suggestion
        pass
```

Reply:
The elipsis is used in context of type stubs. WatchableCommand is just a type-stub to model the following:

```{python}
Callable[[Any, ...], Any]
```

The type-stub nature is signalled by subclassing from ``Protocol``.

Reply:
The related issue is here:
https://github.com/python/typing/issues/109